schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,110 @@ title:              "Post Artifact Request"
 description: "Request a authorization to put and artifact or posting of a URL as an artifact. Note that the `storageType` property is referenced in the response as well."
 type:               object
 oneOf:
+  - title: "Blob Artifact Request"
+    description: |
+      Request a list of requests in a generalized format which can be run to
+      upload an artifact to storage managed by the queue.
+    type: object
+    properties:
+      storageType:
+        description: Artifact storage type, in this case `'blob'`
+        type: string
+        enum:
+          - blob
+      expires:
+        description: |
+          Date-time after which the artifact should be deleted. Note, that
+          these will be collected over time, and artifacts may remain
+          available after expiration. S3 based artifacts are identified in
+          azure table storage and explicitly deleted on S3 after expiration.
+        type: string
+        format: date-time
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type: string
+        maxLength: 255
+      encoding:
```

Comment:
We had a meeting about this and I sent a follow up email.
```
Hi,

I just wanted to send a quick summary of the meeting we had yesterday
regarding content encoding.  The questions we were trying to address
was which content encodings we should support and whether it would be
mandatory or not.

The consensus reached in the meeting was that we would do the following:

1. support arbitrary content encoding in the code but limit it in
configuration to 'identity' and 'gzip' for now

2. declare that the API consumer must support gzip encoding, but that
the response may or may not be gzip encoded

3. build 'automatic' content-encoding into the uploading and
downloading tools so that the common case of using the corresponding
artifact uploading and downloading tooling can transparently handle
and verify that content-encoding is done correctly

4. we did not come to complete agreement about how the automatic
compressing would decide whether or not to do compression, only that
the decision would be made in the uploading task.

We felt that this approach would cover the majority of use cases.
Simple use cases would use our artifact tooling, and more advanced use
cases could opt to handle the rest request directly.  Since
content-encoding is part of the HTTP spec, it should be reasonably
well supported.

We chose to limit to identity (i.e. no encoding) and gzip encoding
because gzip encoding is prevalent on the internet and well supported
in different programming environments.  We are open to adding other
compressors, but wanted to start with a limited subset at the outset.

We discussed various methods for figuring out whether to compress or
not and felt that the best approach would be to build something
minimally smart into the upload tooling which could be overridden,
while also providing the raw rest api for those who wish to have more
control.

This will not happen until the new artifact api is deployed and the
uploading/downloading tools are provided.

If you have any thoughts or concerns about this, we welcome feedback

Thanks,

John
```